File: backend/internal/models/post.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Post struct {
	ID         uuid.UUID `json:"id"`
	AuthorID   uuid.UUID `json:"author_id"`
	AuthorName string    `json:"author_name"`
	Content    string    `json:"content"`
	MediaURL   string    `json:"media_url,omitempty"`
	CreatedAt  time.Time `json:"created_at"`
	UpdatedAt  time.Time `json:"updated_at"`

	LikesCount int  `json:"likes_count"`
	IsLiked    bool `json:"is_liked"`
}

type Like struct {
	PostID    uuid.UUID `json:"post_id"`
	UserID    uuid.UUID `json:"user_id"`
	CreatedAt time.Time `json:"created_at"`
}

type Comment struct {
	ID         uuid.UUID `json:"id"`
	PostID     uuid.UUID `json:"post_id"`
	AuthorID   uuid.UUID `json:"author_id"`
	AuthorName string    `json:"author_name"`
	Content    string    `json:"content"`
	CreatedAt  time.Time `json:"created_at"`
}

File: frontend_mobile/lib/models/post.dart
class Post {
  final String id;
  final String authorId;
  final String? authorName;
  final String content;
  final String? mediaUrl;
  final DateTime createdAt;
  final int likesCount;
  final bool isLiked;

  Post({
    required this.id,
    required this.authorId,
    this.authorName,
    required this.content,
    this.mediaUrl,
    required this.createdAt,
    this.likesCount = 0,
    this.isLiked = false,
  });

  factory Post.fromJson(Map<String, dynamic> json) {
    return Post(
      id: json['id'],
      authorId: json['author_id'],
      authorName: json['author_name'],
      content: json['content'],
      mediaUrl: json['media_url'],
      createdAt: DateTime.parse(json['created_at']),
      likesCount: json['likes_count'] ?? 0,
      isLiked: json['is_liked'] ?? false,
    );
  }
}